### DISCUSSION TITLE
Variables Defined Result in Error for Parser Expression

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#27183] [Q&A General]
```
# Geometry parameters
refinement = 8

[Mesh]
  [ccmg]
    type = ConcentricCircleMeshGenerator
    num_sectors = '${fparse refinement*2}'
    radii = '${circle_radius} ${fparse 1.2*circle_radius}'
    rings = '4 ${refinement} ${refinement}'
    has_outer_square = on
    pitch = ${pitch}
    preserve_volumes = off
    smoothing_max_it = 2
```

for the refinement variable I get the error:
```
No variable 'refinement' found for use in function parser expression.
```
How do I eliminate these types of errors?

**Comment by GiudGiud** [comment]
Hello

The parser could have been thrown off by a missing (other) varaible, like maybe circle_radius?
Or by a runaway quote somewhere else in the input

The way to find these is to make the input smaller and smaller by commenting out blocks

Guillaume

**Reply by TLWise** [reply]
It looks like I have an issue related to my C++ install:

 *** Warning, This code is deprecated and will be removed in future versions:
Please update your main.C to adapt new main function in MOOSE framework, see'test/src/main.C in MOOSE as an example of moose::main()'.

**Reply by GiudGiud** [reply]
you dont have to do that now, we deprecate things smoothly. Worry about it once the other problem is solved.

**Comment by TLWise** [comment]
@GiudGiud I tried this simple case with the following error:
```
  x_min = 0.0
[Mesh]
  type = GeneratedMesh
  dim = 2

  nx = 100
  ny = 100

  xmin = ${x_min}
  xmax = 1.0
  ymin = 0.0
  ymax = 1.0
[]
```
no variable 'x_min' found for substitution expression

Could the problem be related to my OS computer combination?

**Reply by GiudGiud** [reply]
seems like a bug. can you run the diagnostics script in moose/scripts? also what s your version of moose-wasp

**Reply by TLWise** [reply]
What is the syntax for running the diagnostic in moose/scripts, and for determining the version of the moose-wasp?

**Reply by TLWise** [reply]
@GiudGiud I figured out how to run the diagnostic.sh. What should I look for after the run?

**Reply by TLWise** [reply]
Here is the diagnostic output minus the graphics information:
```
ystem Arch: Distributor ID: Ubuntu Description: Ubuntu 22.04.4 LTS Release: 22.04 Codename: jammy

MOOSE Package Version: Custom Build

CPU Count: 8

Memory Free: 14847.438 MB

$CC not set

MPICC WRAPPER NOT FOUND

Python:
	/home/twise/mambaforge3/bin/python
	Python 3.10.13

MODULES NOT AVAILABLE

PETSC_DIR not set

ENVIRONMENT:
_CE_CONDA=
_CE_M=
COLORTERM=truecolor
CONDA_DEFAULT_ENV=base
CONDA_EXE=/home/twise/mambaforge3/bin/conda
CONDA_PREFIX=/home/twise/mambaforge3
CONDA_PROMPT_MODIFIER=(base) 
CONDA_PYTHON_EXE=/home/twise/mambaforge3/bin/python
CONDA_SHLVL=1
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus,guid=073883868a6b4180941c683b65fe053c
DBUS_STARTER_ADDRESS=unix:path=/run/user/1000/bus,guid=073883868a6b4180941c683b65fe053c
DBUS_STARTER_BUS_TYPE=session
DESKTOP_SESSION=ubuntu
DISPLAY=:0
GDMSESSION=ubuntu
GNOME_DESKTOP_SESSION_ID=this-is-deprecated
GNOME_SHELL_SESSION_MODE=ubuntu
GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/d228c499_ed30_4ce5_a163_ea2cc459f97d
GNOME_TERMINAL_SERVICE=:1.389
GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1
GTK_IM_MODULE=ibus
GTK_MODULES=gail:atk-bridge
HOME=/home/twise
IM_CONFIG_PHASE=1
LANG=en_US.UTF-8
LESSCLOSE=/usr/bin/lesspipe %s %s
LESSOPEN=| /usr/bin/lesspipe %s
LOGNAME=twise

PATH=/home/twise/projects/moose/python/peacock:/home/twise/mambaforge3/bin:/home/twise/mambaforge3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
PWD=/home/twise/projects/moose/scripts
QT_ACCESSIBILITY=1
QT_IM_MODULE=ibus
SESSION_MANAGER=local/twise-GS73VR-6RF:@/tmp/.ICE-unix/1846,unix/twise-GS73VR-6RF:/tmp/.ICE-unix/1846
SHELL=/bin/bash
SHLVL=1
SSH_AGENT_LAUNCHER=gnome-keyring
SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
SYSTEMD_EXEC_PID=1846
TERM=xterm-256color
USERNAME=twise
USER=twise
_=/usr/bin/env
VTE_VERSION=6800
WINDOWPATH=2
XAUTHORITY=/run/user/1000/gdm/Xauthority
XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
XDG_CURRENT_DESKTOP=ubuntu:GNOME
XDG_DATA_DIRS=/usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
XDG_MENU_PREFIX=gnome-
XDG_RUNTIME_DIR=/run/user/1000
XDG_SESSION_CLASS=user
XDG_SESSION_DESKTOP=ubuntu
XDG_SESSION_TYPE=x11
XML_CATALOG_FILES=file:///home/twise/mambaforge3/etc/xml/catalog file:///etc/xml/catalog
XMODIFIERS=@im=ibus
```

**Reply by GiudGiud** [reply]
what does `conda list | grep moose` return?

this looks outdated since you are using mambdaforge. but let s be sure

**Reply by TLWise** [reply]
conda list
# packages in environment at /home/twise/mambaforge3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
archspec                  0.2.3              pyhd8ed1ab_0    conda-forge
boltons                   23.1.1             pyhd8ed1ab_0    conda-forge
brotli-python             1.1.0           py310hc6cd4ac_1    conda-forge
brotlipy                  0.7.0           py310h2372a71_1006    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.27.0               hd590300_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py310h2fee648_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     24.1.2          py310hff52083_0    conda-forge
conda-libmamba-solver     24.1.0             pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
cryptography              42.0.5          py310h75e40e8_0    conda-forge
distro                    1.9.0              pyhd8ed1ab_0    conda-forge
fmt                       10.2.1               h00ab1b0_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
jsonpatch                 1.33               pyhd8ed1ab_0    conda-forge
jsonpointer               2.4             py310hff52083_3    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libarchive                3.7.2                h2aa1ff5_1    conda-forge
libcurl                   8.6.0                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_5    conda-forge
libgomp                   13.2.0               h807b86a_5    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libmamba                  1.5.7                had39da4_0    conda-forge
libmambapy                1.5.7           py310h39ff949_0    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsolv                   0.7.28               hfc55251_0    conda-forge
libsqlite                 3.45.2               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_5    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxml2                   2.12.6               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
mamba                     1.5.7           py310h51d5547_0    conda-forge
menuinst                  2.0.2           py310hff52083_0    conda-forge
ncurses                   6.4.20240210         h59595ed_0    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
packaging                 24.0               pyhd8ed1ab_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
platformdirs              4.2.0              pyhd8ed1ab_0    conda-forge
pluggy                    1.4.0              pyhd8ed1ab_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.6           py310h2372a71_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyopenssl                 24.0.0             pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.14         hd12c33a_0_cpython    conda-forge
python_abi                3.10                    4_cp310    conda-forge
readline                  8.2                  h8228510_1    conda-forge
reproc                    14.2.4.post0         hd590300_1    conda-forge
reproc-cpp                14.2.4.post0         h59595ed_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
ruamel.yaml               0.18.6          py310h2372a71_0    conda-forge
ruamel.yaml.clib          0.2.8           py310h2372a71_0    conda-forge
setuptools                69.2.0             pyhd8ed1ab_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
toolz                     0.12.1             pyhd8ed1ab_0    conda-forge
tqdm                      4.66.2             pyhd8ed1ab_0    conda-forge
truststore                0.8.0              pyhd8ed1ab_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml-cpp                  0.8.0                h59595ed_0    conda-forge
zstandard                 0.22.0          py310h1275a96_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge

**Reply by GiudGiud** [reply]
how did you install moose?
which instructions did you follow?

**Reply by TLWise** [reply]
Started here, but that was about one year ago:

https://mooseframework.inl.gov/getting_started/installation/conda.html

It appears the webpage has changed since I performed my reinstall. Initial installation was probably about three years ago.

**Reply by GiudGiud** [reply]
this cant be since you dont have any of the moose packages.
did you forget to run `conda activate moose` before checking the packages?

**Reply by TLWise** [reply]
I don't believe that I ran conda activate moose, prior to running executing the conda command.

**Reply by TLWise** [reply]
Here is the result of the list related to, conda list | grep moose, after executing conda activate moose:

conda list | grep moose
# packages in environment at /home/twise/mambaforge3/envs/moose:
moose-dev                 2024.03.05              build_0    https://conda.software.inl.gov/public
moose-libmesh             2024.02.28              build_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.2.6                   build_8    https://conda.software.inl.gov/public
moose-mpich               4.0.2                  build_15    https://conda.software.inl.gov/public
moose-peacock             2023.11.29           h1bca318_0    https://conda.software.inl.gov/public
moose-petsc               3.20.3                  build_0    https://conda.software.inl.gov/public
moose-tools               2023.12.20           h55675f2_0    https://conda.software.inl.gov/public
moose-wasp                2024.03.01              build_0    https://conda.software.inl.gov/public

**Reply by GiudGiud** [reply]
ok good. This seems up to date enough.
When you run that input file, did you activate moose first?

**Reply by TLWise** [reply]
I run all of my .i files from peacock.

**Reply by GiudGiud** [reply]
I think you still need to do that.
But this might be separate. How do you edit the input file? In Peacock too?

**Reply by TLWise** [reply]
I edit my files in the Ubuntu text editor.

**Comment by TLWise** [comment]
Ran this series of commands:
```
cd ~/projects/moose/scripts
export MOOSE_JOBS=6 METHODS=opt
./update_and_rebuild_petsc.sh --skip-submodule-update --with-packages-download-dir=~/projects/downloads
./update_and_rebuild_libmesh.sh --skip-submodule-update
./update_and_rebuild_wasp.sh --skip-submodule-update
```

Got the following errors:

```
<<< Found PETSc 3.20.3 installation in /home/twise/projects/moose/scripts/../petsc ... >>>
checking whether we can compile a trivial PETSc program... no
checking for TAO support via PETSc... no
configure: error: *** PETSc was not found, but --enable-petsc-required was specified.
Running make -j 6...
./update_and_rebuild_libmesh.sh: line 213: make: command not found
/home/twise/projects/moose/scripts/configure_wasp.sh: line 19: cmake: command not found
Error: configure step for WASP failed to complete successfully
```

**Reply by GiudGiud** [reply]
You are using the conda packages right? 
If so do not run these commands

**Reply by TLWise** [reply]
Are there some preferred commands to use for conda? Now that I have executed these commands, should I take some actions to recover the install to ensure it still works?

**Reply by GiudGiud** [reply]
`mamba update --all` is the preferred path for updating with conda (/mamba formerly)

**Reply by GiudGiud** [reply]
see the other thread, let s see what version you are on
